lib/src/git_backend.rs

```diff
@@ -136,6 +136,29 @@ impl GitBackend {
         Self::init_with_repo(store_path, git_repo_path, git_repo)
     }
 
+    /// Initializes backend by creating a new Git repo at the specified
+    /// workspace path. The workspace directory must exist.
+    pub fn init_colocated(
+        store_path: &Path,
+        workspace_root: &Path,
+    ) -> Result<Self, Box<GitBackendInitError>> {
+        let canonical_workspace_root = {
+            let path = store_path.join(workspace_root);
```

Comment:
TIL `let`s can be blocks with inner `let`s. I actually do this all the time in Haskell, and it makes perfect sense in retrospect, but never thought to try it in Rust...